src/librustc_driver/lib.rs

```diff
@@ -309,17 +309,20 @@ pub fn run_compiler(
                         compiler.output_file().as_ref().map(|p| &**p),
                     );
                 }
-                return sess.compile_status();
+                sess.compile_status()?;
+                return Ok(None);
```

Comment:
I'd prefer `return sess.compile_status().map(|_| None);` for these.

Reply:
Maybe a `let early_exit = || sess.compile_status().map(|_| None);` closure and `return early_exit();` would be cleaner?